application-urlshortener-default/src/main/java/com/xwiki/urlshortener/internal/URLShortenerResourceReferenceHandler.java

```diff
@@ -112,13 +134,28 @@ public void handle(ResourceReference reference, ResourceReferenceHandlerChain ch
         chain.handleNext(reference);
     }
 
+    private List<?> getURLShortenerObjectWithIDAnyWiki(URLShortenerResourceReference resourceReference)
+        throws QueryException
+    {
+        // Note that the query is very slow when solr is reindexing.
```

Comment:
You can rephrase it if needed :)
```suggestion
        // Note that the query is very slow when solr is reindexing.
        // Also, for newly added URLShortener objects, SOLR takes some short type to update with its value. So this SOLR query might also return null if it didn't finish updating the index. 
```